From 5f4dbfcbff75c7ca5e0a88636844f5687253ac61 Mon Sep 17 00:00:00 2001
From: Tim Foster <tim.foster@joyent.com>
Date: Wed, 21 Nov 2018 18:18:54 +0000
Subject: [PATCH] MANTA-4023 mackerel's nodeunit dep cannot install with old
 node Reviewed by: Trent Mick <trentm@gmail.com> Approved by: Trent Mick
 <trentm@gmail.com>

---
 Makefile      | 4 ++--
 package.json  | 2 +-
 test/Makefile | 5 +++--
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 3ad09ea..0835859 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 #
@@ -79,7 +79,7 @@ CLEAN_FILES += $(NODEUNIT) ./node_modules/nodeunit
 
 .PHONY: test
 test: $(NODEUNIT)
-	(cd test && make test)
+	(cd test && NODE_EXEC=../$(NODE_EXEC) make test)
 
 .PHONY: mycheck
 mycheck:
diff --git a/package.json b/package.json
index efe580b..8cc2e00 100644
--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
                 "vasync": "1.3.3"
         },
         "devDependencies": {
-                "nodeunit": "0.7.4",
+                "nodeunit": "0.10.1",
                 "ansidiff": "1.0.0"
         },
         "engines": {
diff --git a/test/Makefile b/test/Makefile
index 6140b94..f7c6ae7 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -5,11 +5,12 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
+NODE_EXEC ?= node
 NODEUNIT=../node_modules/nodeunit/bin/nodeunit
 
 .PHONY: test
 test:
-	find . -name '*.test.js' | xargs -n 1 $(NODEUNIT)
+	find . -name '*.test.js' | xargs -n 1 $(NODE_EXEC) $(NODEUNIT)
-- 
2.21.0

